tests/test_ops.py

```diff
@@ -226,33 +226,49 @@ def test_create_ops_dat_function(self):
             Literal('"u"')
         )
 
-        def test_create_ops_arg_constant(self):
-            a = Constant(name='*a')
+    def test_create_ops_arg_constant(self):
+        a = Constant(name='*a')
 
-            res = create_ops_arg(a, {}, {})
+        res = create_ops_arg(a, {}, {})
 
-            assert res.name == namespace['ops_arg_gbl'].name
-            assert res.args == [
-                Byref(Constant(name='a')),
-                1,
-                Literal('"%s"' % dtype_to_cstr(a.dtype)),
-                namespace['ops_read']
-            ]
+        assert res.name == namespace['ops_arg_gbl'].name
+        assert res.args == [
+            Byref(Constant(name='a')),
+            1,
+            Literal('"%s"' % dtype_to_cstr(a.dtype)),
+            namespace['ops_read']
+        ]
+
+    @pytest.mark.parametrize('read', [True, False])
+    def test_create_ops_arg_function(self, read):
+        u = OpsAccessible('u', np.float32, read)
 
-        @pytest.mark.parametrize('read', [True, False])
-        def test_create_ops_arg_function(self, read):
-            u = OpsAccessible('u', np.float32, read)
+        dat = OpsDat('u_dat')
+        stencil = OpsStencil('stencil')
 
-            dat = OpsDat('u_dat')
-            stencil = OpsStencil('stencil')
+        res = create_ops_arg(u, {'u': dat}, {u: stencil})
 
-            res = create_ops_arg(u, {'u': dat}, {u: stencil})
+        assert res.name == namespace['ops_arg_dat'].name
+        assert res.args == [
+            dat,
+            1,
+            stencil,
+            Literal('"%s"' % dtype_to_cstr(u.dtype)),
+            namespace['ops_read'] if read else namespace['ops_write']
+        ]
+
+    @pytest.mark.parametrize('equation, expected', [
+        ('Eq(u.forward, u.dt2 + u.dxr - u.dyr - u.dyl)',
+            'ops_block block = ops_decl_block(2, "block");'),
+        ('Eq(u.forward,u+1)',
+            'ops_block block = ops_decl_block(2, "block");')
+    ])
+    def test_create_ops_block(self, equation, expected):
+        """
+        Test if ops_block has been successfully generated
+        """
+        grid_2d = Grid(shape=(4, 4))
+        u = TimeFunction(name='u', grid=grid_2d, time_order=2, save=Buffer(10))  # noqa
+        op = OperatorOPS(eval(equation))
```

Comment:
I am curious why you didn't use just `Operator`. 

I mean, when using Devito we will only specify the backend through the `DEVITO_BACKEND` env. variable. And then, Devito will choose the correct `Operator`  to use.  
So if you test using this command: `DEVITO_BACKEND=ops pytest tests/test_ops.py` it will switch to this Operator automaticaly.

Reply:
actually i just followed the pattern from `test_kernel_generation`.
u're right thou, let me change both